app/scripts/modules/netflix/pipeline/stage/canary/canaryStage.js

```diff
@@ -42,27 +123,51 @@ module.exports = angular.module('spinnaker.netflix.pipeline.stage.canaryStage',
-    $scope.stage.canary.owner = $scope.stage.canary.owner || (user.authenticated ? user.name : null);
-    $scope.stage.canary.watchers = $scope.stage.canary.watchers || [];
-    $scope.stage.canary.canaryConfig = $scope.stage.canary.canaryConfig || { name: [$scope.pipeline.name, 'Canary'].join(' - ') };
-    $scope.stage.canary.canaryConfig.canaryAnalysisConfig = $scope.stage.canary.canaryConfig.canaryAnalysisConfig || {};
-    $scope.stage.canary.canaryConfig.canaryAnalysisConfig.notificationHours = $scope.stage.canary.canaryConfig.canaryAnalysisConfig.notificationHours || [];
-    $scope.stage.canary.canaryConfig.canaryAnalysisConfig.useLookback = $scope.stage.canary.canaryConfig.canaryAnalysisConfig.useLookback || false;
-    $scope.stage.canary.canaryConfig.canaryAnalysisConfig.lookbackMins = $scope.stage.canary.canaryConfig.canaryAnalysisConfig.lookbackMins || 0;
-    $scope.stage.canary.canaryConfig.canaryAnalysisConfig.useGlobalDataset = $scope.stage.canary.canaryConfig.canaryAnalysisConfig.useGlobalDataset || false;
-    $scope.stage.canary.canaryConfig.actionsForUnhealthyCanary = $scope.stage.canary.canaryConfig.actionsForUnhealthyCanary || [
-      {action: 'DISABLE'},
-      {action: 'TERMINATE', delayBeforeActionInMins: 60}
-    ];
+    stage.baseline = stage.baseline || {};
+    stage.scaleUp = stage.scaleUp || {};
 
-    let overriddenCloudProvider = 'aws';
+    stage.canary = stage.canary || {};
+    stage.canary.owner = stage.canary.owner || (user.authenticated ? user.name : null);
+    stage.canary.watchers = stage.canary.watchers || [];
+
+    stage.canary.canaryConfig = stage.canary.canaryConfig || {name: [$scope.pipeline.name, 'Canary'].join(' - ')};
+    stage.canary.canaryConfig.canaryAnalysisConfig = stage.canary.canaryConfig.canaryAnalysisConfig || {};
+
+    if (!stage.scaleUp) {
+      stage.scaleUp.enabled = false;
+    }
+
+    const cc = stage.canary.canaryConfig;
+    if (cc.actionsForUnhealthyCanary && cc.actionsForUnhealthyCanary.find(action => action.action === 'TERMINATE')) {
+      this.terminateUnhealthyCanaryEnabled = true;
+    }
 
+    const cac = cc.canaryAnalysisConfig;
+    if (cac.useLookback) {
+      this.analysisType = 'SLIDING_LOOKBACK';
+    } else {
+      this.analysisType = 'GROWING';
+    }
+
+    let overriddenCloudProvider = 'aws';
     if ($scope.stage.isNew) {
+
+      // apply defaults
+      cc.lifetimeHours = 3;
+      cac.combinedCanaryResultStrategy = 'AGGREGATE';
+      set(cc, 'canaryHealthCheckHandler.minimumCanaryResultScore', 75);
+      set(cc, 'canarySuccessCriteria.canaryResultScore', 95);
+      cac.notificationHours = [1, 2, 3];
+      cac.canaryAnalysisIntervalMins = 30;
+      cac.useLookback = false;
+      cac.lookbackMins = 0;
+      cac.useGlobalDataset = false;
+      cc.actionsForUnhealthyCanary = [{action: 'DISABLE'}, {action: 'TERMINATE', delayBeforeActionInMins: 60}];
+      this.terminateUnhealthyCanaryEnabled = true;
+      cac.beginCanaryAnalysisAfterMins = 0;
```

Comment:
It might read easier if the defaults were set as a new object here, rather than a field at a time, e.g.
```javascript
this.terminateUnhealthyCanaryEnabled = true;
scope.stage.canaryConfig = {
  lifetimeHours: 3,
  canaryHealthCheckHandler: {
    minimumCanaryResultScore: 75
  },
  canarySuccessCriteria: {
    canaryResultScore: 95
  },
  actionsForUnhealthyCanary: [
    { action: 'DISABLE' },
    { action: 'TERMINATE, delayBeforeActionInMins: 60 }
  ],
  canaryAnalysisConfig: {
    combinedCanaryResultStrategy: 'AGGREGATE',
    notificationHours: [1, 2, 3],
    canaryAnalysisIntervalMins: 30,
    useLookback: false,
    lookbackMins: 0,
    useGlobalDataset: false,
    beginCanaryAnalysisAfterMins: 0
  }
};
```